packages-internal/test-utils/src/describeConformance.tsx

```diff
@@ -91,7 +91,7 @@ function throwMissingPropError(field: string): never {
  * the root component.
  */
 export function testClassName(
-  element: React.ReactElement<any>,
+  element: React.ReactElement<HTMLElement & { 'data-testid'?: string }>,
```

Comment:
All changes of `any` to `unknown` in this PR are straight forward, except for this file.

Reply:
Could you explain the criteria you used to choose the type of `element` in the different functions? I'm trying to understand the impact it will have on the `describeConformance` usage. Will elements passed to `testClassName` without the `'data-tested'` prop throw a type error? Where would that error show?

Reply:
You are right, calling testClassName without passing `data-testid` is throwing type error. 

This is tricky to fix, `data-testid` is added to `element` as below. 

https://github.com/mui/material-ui/blob/bb620d85a89651ca4350e8861bbb35776d583726/packages-internal/test-utils/src/describeConformance.tsx#L108

I think we have 2 options to fix

1. add `data-testid` type to argument as above, but it throws type error when consumers of this function doesn't add `data-testid` type
2. Keep `element` argument type as `React.ReactElement<unknown>` and type cast `element` type near it's usage as shown below.

> ```diff
>  React.cloneElement(element
> + as React.ReactElement<unknown & {[data-testid]:string}>
>, { className, 'data-testid': testId }),
>  
> ```

By going with option 2 , we can keep types of consumers of this function to be flexible. what do you think?

What i did for now is, i've reverted all changes in this file [here](https://github.com/mui/material-ui/pull/43402/commits/7d3d548cae8dff177b5759f783b3ec550e9254ee), what ever option we choose i'll create a seprate PR as this PR is alread too big and i don't want this descion to block this PR from merging.

Reply:
Let's open another PR and discuss it there 👌🏼 

I think we can create a type that is the same as ReactElement but supports `data-` attributes